src/main/java/sk/freemap/gpxAnimator/ui/MainFrame.java

```diff
@@ -400,6 +407,42 @@ protected void configurationChanged() {
     panel.add(addTrackButton, gbc_addTrackButton);
     addTrackButton.addActionListener(addTrackActionListener);
 
+
+    resetGeneralSettings = e -> {
+      try {
+        generalSettingsPanel.updateProperty(8,30);
+      } catch (IOException ex) {
+        ex.printStackTrace();
+      }
+    };
+
+    saveGeneralSettings = e -> {
+      try {
+        generalSettingsPanel.updateProperty(-1,-1);
+      } catch (IOException ex) {
+        ex.printStackTrace();
+      }
+    };
+
+
+    final JButton resetSettings = new JButton("Reset settings");
+    final GridBagConstraints resetSettingsButton = new GridBagConstraints();
+    resetSettingsButton.anchor = GridBagConstraints.NORTHWEST;
+    resetSettingsButton.insets = new Insets(0, 0, 0, 5);
+    resetSettingsButton.gridx = 2;
+    resetSettingsButton.gridy = 0;
+    panel.add(resetSettings, resetSettingsButton);
+    addTrackButton.addActionListener(resetGeneralSettings);
+
+    final JButton saveSettings = new JButton("Save settings");
+    final GridBagConstraints saveSettingsButton = new GridBagConstraints();
+    saveSettingsButton.anchor = GridBagConstraints.NORTHWEST;
+    saveSettingsButton.insets = new Insets(0, 0, 0, 5);
+    saveSettingsButton.gridx = 3;
+    saveSettingsButton.gridy = 0;
+    panel.add(saveSettings, saveSettingsButton);
+    addTrackButton.addActionListener(saveGeneralSettings);
```

Comment:
I guess you want to add the action listener to the `saveSettingsButton`, not the `addTrackButton`. :-)